lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -497,10 +494,7 @@ export class UserAgentApplication {
           this._cacheStorage.setItem(Constants.msalError, "");
           this._cacheStorage.setItem(Constants.msalErrorDescription, "");
           const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-          if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-              this._cacheStorage.setItem(authorityKey, this.authority);
-          }
-
+          this._cacheStorage.setItem(authorityKey, this.authority);
```

Comment:
are we missing the storeAuthStateInCookie flag here?

Reply:
No, we only need this flag for redirect functions like loginRedirect and acquireTokenRedirect.

Reply:
We saw some issues earlier when storeAuthStateInCookie flag was added to only loginRedirect and acquireTokenRedirect. We are currently passing this flag for popup cases as well. I think we should do it everywhere for consistency.